src/Mod/Path/PathScripts/post/grbl_post.py

```diff
@@ -285,52 +285,50 @@ def export(objectslist, filename, argstring):
     # turn coolant on if required
     if OUTPUT_COMMENTS:
         if not coolantMode == 'None':
-            gcode += linenumber() + '(Coolant On:' + coolantMode + ')\n'
+            gcode.append( linenumber() + '(Coolant On:' + coolantMode + ')\n' )
     if coolantMode == 'Flood':
-        gcode  += linenumber() + 'M8' + '\n'
+        gcode.append( linenumber() + 'M8' + '\n' )
     if coolantMode == 'Mist':
-        gcode += linenumber() + 'M7' + '\n'
+        gcode.append( linenumber() + 'M7' + '\n' )
 
     # Parse the op
-    gcode += parse(obj)
+    gcode.append( parse(obj)  )
 
     # do the post_op
     if OUTPUT_COMMENTS:
-      gcode += linenumber() + "(Finish operation: " + obj.Label + ")\n"
+      gcode.append( linenumber() + "(Finish operation: " + obj.Label + ")\n" )
     for line in POST_OPERATION.splitlines(True):
-      gcode += linenumber() + line
+      gcode.append( linenumber() + line )
 
     # turn coolant off if required
     if not coolantMode == 'None':
         if OUTPUT_COMMENTS:
-            gcode += linenumber() + '(Coolant Off:' + coolantMode + ')\n'
-        gcode += linenumber() +'M9' + '\n'
+            gcode.append( linenumber() + '(Coolant Off:' + coolantMode + ')\n' )
+        gcode.append( linenumber() +'M9' + '\n' )
 
   # do the post_amble
   if OUTPUT_BCNC:
-    gcode += linenumber() + "(Block-name: post_amble)\n"
-    gcode += linenumber() + "(Block-expand: 0)\n"
-    gcode += linenumber() + "(Block-enable: 1)\n"
+    gcode.append( linenumber() + "(Block-name: post_amble)\n" )
+    gcode.append( linenumber() + "(Block-expand: 0)\n" )
+    gcode.append( linenumber() + "(Block-enable: 1)\n" )
   if OUTPUT_COMMENTS:
-    gcode += linenumber() + "(Begin postamble)\n"
+    gcode.append( linenumber() + "(Begin postamble)\n" )
   for line in POSTAMBLE.splitlines(True):
-    gcode += linenumber() + line
+    gcode.append( linenumber() + line )
 
   if RETURN_TO:
-    gcode += linenumber() + "G0 X%s Y%s" % tuple(RETURN_TO)
+    gcode.append( linenumber() + "G0 X%s Y%s" % tuple(RETURN_TO) )
 
   # show the gCode result dialog
+  sep=''
+  final = sep.join(gcode)
```

Comment:
if you go this route, wouldn't it be more efficient to remove the `\n` from all lines and use that as the separator?
just a thought.